Source/Particles/Collision/BinaryCollision/DSMC/SplitAndScatterFunc.H

```diff
@@ -235,6 +313,184 @@ public:
+                auto& uy1 = soa_products_data[0].m_rdata[PIdx::uy][species1_index];
+                auto& uz1 = soa_products_data[0].m_rdata[PIdx::uz][species1_index];
+                auto& ux_p1 = soa_products_data[ioniz_product1_list_index].m_rdata[PIdx::ux][product1_index];
+                auto& uy_p1 = soa_products_data[ioniz_product1_list_index].m_rdata[PIdx::uy][product1_index];
+                auto& uz_p1 = soa_products_data[ioniz_product1_list_index].m_rdata[PIdx::uz][product1_index];
+                auto& ux_p2 = soa_products_data[ioniz_product2_list_index].m_rdata[PIdx::ux][product2_index];
+                auto& uy_p2 = soa_products_data[ioniz_product2_list_index].m_rdata[PIdx::uy][product2_index];
+                auto& uz_p2 = soa_products_data[ioniz_product2_list_index].m_rdata[PIdx::uz][product2_index];
+
+#if (defined WARPX_DIM_RZ)
+                /* In RZ geometry, macroparticles can collide with other macroparticles
+                * in the same *cylindrical* cell. For this reason, collisions between macroparticles
+                * are actually not local in space. In this case, the underlying assumption is that
+                * particles within the same cylindrical cell represent a cylindrically-symmetry
+                * momentum distribution function. Therefore, here, we temporarily rotate the
+                * momentum of one of the macroparticles in agreement with this cylindrical symmetry.
+                * (This is technically only valid if we use only the m=0 azimuthal mode in the simulation;
+                * there is a corresponding assert statement at initialization.)
+                */
+                amrex::ParticleReal const theta = (
+                    soa_products_data[ioniz_product1_list_index].m_rdata[PIdx::theta][product1_index]
+                    - soa_products_data[0].m_rdata[PIdx::theta][species1_index]
+                );
+                amrex::ParticleReal const ux1buf = ux1;
+                ux1 = ux1buf*std::cos(theta) - uy1*std::sin(theta);
+                uy1 = ux1buf*std::sin(theta) + uy1*std::cos(theta);
+#endif
+
+                // for simplicity (for now) we assume non-relativistic particles
+                // and simply calculate the center-of-momentum velocity from the
+                // rest masses
+                auto const uCOM_x = (m1 * ux1 + m2 * ux_p2) / (m1 + m2);
+                auto const uCOM_y = (m1 * uy1 + m2 * uy_p2) / (m1 + m2);
+                auto const uCOM_z = (m1 * uz1 + m2 * uz_p2) / (m1 + m2);
+
+                // transform to COM frame
+                ux1 -= uCOM_x;
+                uy1 -= uCOM_y;
+                uz1 -= uCOM_z;
+                ux_p1 -= uCOM_x;
+                uy_p1 -= uCOM_y;
+                uz_p1 -= uCOM_z;
+                ux_p2 -= uCOM_x;
+                uy_p2 -= uCOM_y;
+                uz_p2 -= uCOM_z;
+
+                if (mask[i] == int(ScatteringProcessType::IONIZATION)) {
```

Comment:
This seems redundant: we are already in an `else if` condition for `(mask[i] == int(ScatteringProcessType::IONIZATION))`

Reply:
I added it in as a forward-looking thing since we could add excitation events (or other product producing events). I know we will need to change the higher if condition for this as well in that case, but it would still clearly show where the logic should split.